website/src/pages/downloads/linux.tsx

```diff
@@ -1,81 +1,22 @@
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import { usePluginData } from '@docusaurus/useGlobalData';
 import { faLinux } from '@fortawesome/free-brands-svg-icons';
 import { faDownload, faPaste, faTerminal } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import TailWindThemeSelector from '@site/src/components/TailWindThemeSelector';
 import { TelemetryLink } from '@site/src/components/TelemetryLink';
+import type { GitHubMetadata } from '@site/src/plugins/GitHubMetadata';
 import Layout from '@theme/Layout';
-import type { Dispatch, SetStateAction } from 'react';
-import React, { useEffect, useState } from 'react';
-
-async function grabFilenameForLinux(
-  setDownloadData: Dispatch<
-    SetStateAction<{
-      version: string;
-      flatpak: string;
-      amd64: string;
-      arm64: string;
-    }>
-  >,
-): Promise<void> {
-  const result = await fetch('https://api.github.com/repos/containers/podman-desktop/releases/latest');
-  const jsonContent = await result.json();
-  const {
-    tag_name,
-    name: rawName,
-    assets,
-  } = jsonContent as {
-    tag_name?: string;
-    name: string;
-    assets: Array<{ name: string; browser_download_url: string }>;
-  };
-
-  const version = (tag_name ?? rawName).replace(/^v/, '');
-
-  const flatpakAssets = assets.filter(a => a.name.endsWith('.flatpak'));
-  if (flatpakAssets.length !== 1) {
-    throw new Error('Unable to grab .flatpak');
-  }
-  const flatpakLink = flatpakAssets[0];
-
-  const armAssets = assets.filter(a => a.name.endsWith('-arm64.tar.gz'));
-  if (armAssets.length !== 1) {
-    throw new Error('Unable to grab ARM64 (.tar.gz)');
-  }
-  const armLink = armAssets[0];
-
-  const amdAssets = assets.filter(a => a.name.endsWith('.tar.gz') && !a.name.includes('arm64'));
-  if (amdAssets.length !== 1) {
-    throw new Error('Unable to grab AMD64 (.tar.gz)');
-  }
-  const amdLink = amdAssets[0];
-
-  setDownloadData({
-    version,
-    flatpak: flatpakLink.browser_download_url,
-    arm64: armLink.browser_download_url,
-    amd64: amdLink.browser_download_url,
-  });
-}
+import React from 'react';
 
 export function LinuxDownloads(): JSX.Element {
-  const [downloadData, setDownloadData] = useState({
-    version: '',
-    arm64: '',
-    amd64: '',
-    flatpak: '',
-  });
+  // Use the generic version of usePluginData for full type safety
+  const { linuxDownloads } = usePluginData('docusaurus-plugin-github-metadata') as GitHubMetadata;
```

Comment:
```suggestion
  // Use the generic version of usePluginData for full type safety
  const { linuxDownloads } = usePluginData('docusaurus-plugin-github-metadata') as GitHubMetadata;
```

I commented somewhere else but it should not use `as GithubMetadata` but the generics but really use it 
also the comment should be dropped as the comment is not doing what is expected

```ts
  const { linuxDownloads } = usePluginData<GitHubMetadata>('docusaurus-plugin-github-metadata');
```

Reply:
Yes the comment is wrong, I tried to do this at first too but I have an error that I am unable to solve:

<img width="731" height="336" alt="Screenshot 2025-07-16 at 10 00 33" src="https://github.com/user-attachments/assets/2e1e0c39-9c98-4919-a302-f8c4ee3dac53" />